### PR TITLE
expose native Fn `conv2dBackwardData` in `flashlight_binding.cc`

### PR DESCRIPTION
**Pull request by cryptodeal**
Add `conv2dBackwardData` to `flashlight_binding.cc` & update `ffi_tensor.ts` accordingly; remaining changes are purely enforcing formatting standards w `bun format`.

### CODE REVIEW

File: shumai/cpp/flashlight_binding.cc [hunk 1]
```diff
@@ -90,6 +95,34 @@ JSTypedArrayBytesDeallocator genTensorDestroyer() {
...
+  auto* used_wt = reinterpret_cast<fl::Tensor*>(wt);
+
+  auto payload = std::make_shared<fl::detail::AutogradPayload>();
+  std::shared_ptr<fl::DynamicBenchmark> dataBench;
+  auto result =
+      used_in->backend()
+          .getExtension<fl::AutogradExtension>()
+          .conv2dBackwardData(*used_grad_in, *used_in, *used_wt, sx, sy, px, py,
+                              dx, dy, groups, dataBench, payload);
+  return new fl::Tensor(result);
```
Comment (bwasti) [inline shumai/cpp/flashlight_binding.cc:123]: before returning, always update the byte tracker `g_bytes_used += result.bytes();`

File: shumai/cpp/flashlight_binding.cc [hunk 2]
```diff
@@ -90,6 +95,34 @@ JSTypedArrayBytesDeallocator genTensorDestroyer() {
   return destroyTensor;
 }
 
+// `grad_in` is Shumai equivalent for Flashlight `gradOutput`
+void* conv2dBackwardData(void* grad_in,
```
Comment (bwasti) [inline shumai/cpp/flashlight_binding.cc:99]: please prefix with `_` (convention for functions that take in a `Tensor`) and move to the bottom of the file (after `_copy`)

File: shumai/cpp/flashlight_binding.cc (hunk 1 above)
Comment (bwasti) [reply shumai/cpp/flashlight_binding.cc:123]: I really should set up a convention here... e.g. `RETURN_TENSOR(result)` and that does all the logic we need (like `LOCK_GUARD`)

Comment (cryptodeal) [reply shumai/cpp/flashlight_binding.cc:123]: Yea, that convention would probably keep the code a bit cleaner, so I'm a proponent of adding `RETURN_TENSOR`!

File: shumai/cpp/flashlight_binding.cc (hunk 2 above)
Comment (cryptodeal) [reply shumai/cpp/flashlight_binding.cc:99]: Fixed!

File: shumai/cpp/flashlight_binding.cc [hunk 3]
```diff
@@ -1,7 +1,12 @@
+
 #include <atomic>
 #include <cstdlib>
 #include <iostream>
 #include "flashlight/fl/autograd/Functions.h"
+#include "flashlight/fl/autograd/tensor/AutogradExtension.h"
+#include "flashlight/fl/autograd/tensor/AutogradExtensionBackends.h"
```
[the remark this answers is not in the document]
Comment (cryptodeal) [reply shumai/cpp/flashlight_binding.cc:7]: Removing the `AutogradExtensionBackends.h` include resulted in build errors:


```sh
Consolidate compiler generated dependencies of target flashlight_binding
[ 50%] Building CXX object CMakeFiles/flashlight_binding.dir/shumai/cpp/flashlight_binding.cc.o
/Users/cryptodeal/shumai/shumai/cpp/flashlight_binding.cc:288:11: error: member access into incomplete type 'fl::TensorBackend'
          .getExtension<fl::AutogradExtension>()
          ^
/Users/cryptodeal/usr/include/flashlight/fl/tensor/TensorBase.h:36:7: note: forward declaration of 'fl::TensorBackend'
class TensorBackend;
      ^
1 error generated.
make[2]: *** [CMakeFiles/flashlight_binding.dir/build.make:76: CMakeFiles/flashlight_binding.dir/shumai/cpp/flashlight_binding.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/flashlight_binding.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

So this appears to be a necessary include.

**Pull request merged**